CONTRIBUTING.md

```diff
@@ -1,13 +1,608 @@
+to contribute, and how. This sets you up for success - you can avoid duplicating
+an effort that may already be underway, adding a feature that may be rejected,
+or heading down a path that you would be steered away from at review time. The
+best way to communicate your intent is via a detailed GitHub issue. Take a look
+first to see if there's already an issue relating to the thing you'd like to
+contribute. If there isn't, please raise a new one! Let us know what you'd like
+to work on, and why. SIG Release members can't always triage new issues
+immediately, but we encourage you to bring them to our attention via one of our
+[communication channels][contact].
+
+> NOTE: Unlike the core [Kubernetes release cycle][kubernetes-release-cycle],
+> [SIG Release repositories][sig-release-subprojects] are versioned/released on
+> ad hoc (as needed basis), which means that the changes you've made will
+> not necessarily be immediately available for consumption in tagged artifacts.
+>
+> We may release components for the following, non-exhaustive reasons:
+>
+> - unblocking a core Kubernetes release cycle
+> - unblocking Kubernetes projects that leverage our libraries
+> - mitigating a security vulnerability
+> - critical bug fixes or regressions
+> - a feature is incomplete or does not work as intended
+>
+> If you need a new component release to unblock your project, please contact us to request one.
+
+<!--
+TODO(contribute): Review Crossplane release process for more inspiration
+ref: https://github.com/crossplane/crossplane/blob/master/docs/reference/release-cycle.md
+-->
+
+Be sure to practice [good git commit hygiene] as you make your changes. All but
+the smallest changes should be broken up into a few commits that tell a story.
+Use your git commits to provide context for the folks who will review PR, and
+the folks who will be spelunking the codebase in the months and years to come.
+Ensure each of your commits is signed-off by using `git commit -s`. SIG Release
+members highly values readable, idiomatic Go code. Familiarize yourself with
+the [Coding Style](#coding-style) section below and try to preempt any comments
+your reviewers would otherwise leave. Most repos will contain `Makefile`s or
+[`mage`][mage] targets that will help you build/test/lint/verify your changes.
+If you don't see a target for your use case, maybe that could be your first
+contribution!
+
+<!--
+TODO(contribute): Decide on stance around testing libraries: https://github.com/golang/go/wiki/TestComments#assert-libraries
+-->
+
+All code should be covered by tests. We strive, wherever possible, to follow
+the common Go [test review comments]. We encourage the use of table-driven
+unit tests - you can find an example [below](#prefer-table-driven-tests). Note
+that when opening a PR your reviewer will expect you to detail how you've
+tested your work. For all but the smallest changes, some manual testing is
+encouraged in addition to unit tests.
```

Comment:
There are changes to some SIG Release projects that cannot be tested unless you have speical permissions (e.g. changes to krel, promoter tools...). I'm wondering should we word it like that -- that manual testing is highly encouraged, but that it might not be always possible, and in that case, a SIG Release maintainer will help you with that.